src/libs/dotcms.api.js

```diff
@@ -2,10 +2,32 @@ const { login, isLogin, logout } = require('./dotcms.auth');
 const { getCurrentSite } = require('./dotcms.sites');
 const { esSearch } = require('./dotcms.esSearch');
 const fetch = require('node-fetch');
+let languagesConf;
+let language = {
+    code: '',
+    id: ''
+};
+
+const setLanguage = (pathname) => {
+    language.code = pathname.split('/')[1];
+    const langId = languagesConf.find((lang) => lang.languageCode === language.code);
+    if (langId) {
+        language.id = langId.id;
+    } else {
+        language.code = null;
+        language.id = 1;
+    }
+    return language;
+};
 
 const getUrl = (pathname) => {
+    if (language.code) {
+        let pathnameSplitted = pathname.split('/');
+        pathnameSplitted.splice(1, 1);
+        pathname = pathnameSplitted.join('/');
+    }
     const host = process.env.NODE_ENV !== 'development' ? process.env.REACT_APP_DOTCMS_HOST : '';
-    return `${host}/api/v1/page/json/${pathname.slice(1)}?language_id=1`;
+    return `${host}/api/v1/page/json/${pathname.slice(1)}?language_id=${language.id}`;
```

Comment:
Keeping the state of the lang is the library is not a good idea, this should be kept in the app, the library should only receive the lang as a param.

This library is a external resource, it will be installed by npm in any SPA, it should not contain any SPA state or SPA state related data.